schema/elements.go

```diff
@@ -77,6 +79,8 @@ const (
 	// Separable means the items of the container type have no particular
 	// relationship (default behavior for maps and structs).
 	Separable = ElementRelationship("separable")
+	// Lookup only applies to untyped (see the documentation there).
```

Comment:
This semantic is a little sloppy, you can't look at the schema and now what it's going to do

Reply:
Since `deduceType`  is the only lookup method you have, could this be named `Deduce` instead?